src/harness/reference_models/ppa_model/ppa_ref_model.py

```diff
@@ -0,0 +1,82 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from concurrent.futures import ProcessPoolExecutor as Pool
+import multiprocessing
+from utils import *
+from compute import load_partials
+from reference_models.geo.vincenty import GeodesicPoint
+
+THRESHOLD = -96
+
+
+def __compute_points_gain_prop(device):
+  pool = Pool(multiprocessing.cpu_count())
+  install_param = device['installationParam']
+  # Get all the Function Partials
+  great_circle_partial, antenna_gain_partial, \
+  prop_partial, cnt_partial = load_partials(install_param, THRESHOLD)
+  # Compute all the Points in 0-359 every 200m upto 40km
+  points = pool.map(great_circle_partial, range(0, 360))
+  # Compute the Gain for all Direction
+  antenna_gain = pool.map(antenna_gain_partial, range(0, 360))
+  # Compute the Path Loss
+  path_loss = pool.map(prop_partial, enumerate(points))
```

Comment:
Only this one is a bit heavy in terms of calculation, other are very fast and the multiprocessing may in fact slow things down. See my previous comment.